snapshot/git/gitdb/bundlestore.go

```diff
@@ -0,0 +1,200 @@
+package gitdb
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+
+	snap "github.com/scootdev/scoot/snapshot"
+	"github.com/scootdev/scoot/snapshot/bundlestore"
+)
+
+// BundlestoreConfig defines how to talk to Bundlestore
+type BundlestoreConfig struct {
+	Store bundlestore.Store
+}
+
+type bundlestoreBackend struct {
+	cfg *BundlestoreConfig
+}
+
+const bundlestoreIDText = "bs"
+
+// "bs-gc-<bundle>-<stream>-<sha>"
+
+func (b *bundlestoreBackend) parseID(id snap.ID, kind snapshotKind, extraParts []string) (snapshot, error) {
+	if len(extraParts) != 3 {
+		return nil, fmt.Errorf("cannot parse snapshot ID: expected 5 parts in bundlestore id: %s", id)
+	}
+	bundleName, streamName, sha := extraParts[0], extraParts[1], extraParts[2]
+
+	if err := validSha(sha); err != nil {
+		return nil, err
+	}
+
+	return &bundlestoreSnapshot{kind: kind, sha: sha, bundleName: bundleName, streamName: streamName}, nil
+}
+
+func (b *bundlestoreBackend) upload(s snapshot, db *DB) (snapshot, error) {
+	// We only have to upload a localSnapshot
+	switch s := s.(type) {
+	case *tagsSnapshot:
+		return s, nil
+	case *streamSnapshot:
```

Comment:
A stream updates; tags don't.

Commits in a stream are related; commits in different tags aren't.

This means that you can just update a stream.

You create a streamSnapshot by pushing something new to master, then creating the ID, and using that.